zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/InterpreterSettingManager.java

```diff
@@ -1035,6 +1035,9 @@ private boolean inlineRemove(String id, boolean initiator) throws IOException {
     }
 
     File localRepoDir = new File(conf.getInterpreterLocalRepoPath() + "/" + id);
+    if (localRepoDir.getAbsolutePath().contains("..")) {
```

Comment:
Any idea why deleting the localRepoDir is not part of the if block?

Reply:
I also wonder which APIs or methods are affected by this logic? I think we'd better have tests for this change.

Reply:
Normal access should not allow him to be able to use... / to access unintended directories

Reply:
@iiiusky Ah, do you mean normal users should not call with the id including `..` and so on?

Reply:
@jongyoul Yes, or it will filter .. all internally, and then go to a specific directory to start concatenating the final path, instead of completely trusting the string path passed in by the user

Reply:
> Any idea why deleting the localRepoDir is not part of the if block?

Let me clarify my statement.
Any idea why deleting the localRepoDir is not part of the if block `if (interpreterSettings.containsKey(id)) {` which is directly above this change?

Reply:
Agree with @Reamer , put it into if block will fix this issue as well.